cases/IFAD_Sudan/quantile_feature_extractor.py

```diff
@@ -0,0 +1,149 @@
+"""This file provides the example of a basic feature extractor that computes
+indices from preprocessed data and extracts quantiles from those indices.
+
+Implementing a function in a source file, and then calling the
+`apply_feature_extractor` in OpenEO should be enoug to run the inference.
+"""
+import openeo
+import xarray as xr
+
+# Spatial and temporal definitions
+from openeo_gfmap import BoundingBoxExtent, TemporalContext
+
+# Backend and context
+from openeo_gfmap.backend import Backend, BackendContext
+from openeo_gfmap.features import PatchFeatureExtractor
+from openeo_gfmap.features.feature_extractor import apply_feature_extractor
+
+# Fetching type (either TILE, POLYGON or POINT)
+from openeo_gfmap.fetching import FetchType, build_sentinel2_l2a_extractor
+
+# Preprocessing options
+from openeo_gfmap.preprocessing import (
+    linear_interpolation,
+    mask_scl_dilation,
+    median_compositing,
+)
+
+
+class QuantileIndicesExtractor(PatchFeatureExtractor):
+    """Performs feature extraction by returning qunatile indices of the input
+    array."""
+
+    def output_labels(self) -> list:
+        indices_names = ["NDVI", "NDWI", "NDMI", "NDRE", "NDRE5", "B11", "B12"]
+        quantile_values = ["10", "50", "90", "IQR"]
+        quantile_names = [
+            index_name + ":" + quantile_value
+            for index_name in indices_names
+            for quantile_value in quantile_values
+        ]
+        return quantile_names
+
+    def execute(self, inarr: xr.DataArray) -> xr.DataArray:
+        # compute indices
+        b03 = inarr.sel(bands="S2-B03")
+        b04 = inarr.sel(bands="S2-B04")
+        b05 = inarr.sel(bands="S2-B05")
+        b06 = inarr.sel(bands="S2-B06")
+        b08 = inarr.sel(bands="S2-B08")
+        b11 = inarr.sel(bands="S2-B11")
+        b12 = inarr.sel(bands="S2-B12")
+
+        ndvi = (b08 - b04) / (b08 + b04)
+        ndwi = (b03 - b08) / (b03 + b08)
+        ndmi = (b08 - b11) / (b08 + b11)
+        ndre = (b05 - b08) / (b05 + b08)
+        ndre5 = (b06 - b08) / (b06 + b08)
+
+        indices = [ndvi, ndwi, ndmi, ndre, ndre5, b11, b12]
+
+        quantile_arrays = []
+
+        for data in indices:
+            q01 = data.quantile(0.1, dim=["t"]).drop("quantile")
+            q05 = data.quantile(0.5, dim=["t"]).drop("quantile")
+            q09 = data.quantile(0.9, dim=["t"]).drop("quantile")
+            iqr = q09 - q01
+
+            quantile_arrays.extend([q01, q05, q09, iqr])
+
+        # Pack the quantile arrays into a single array
+        quantile_array = (
+            xr.concat(quantile_arrays, dim="bands")
+            .assign_coords({"bands": self.output_labels()})
+            .transpose("bands", "y", "x")
+        )
+
+        return quantile_array
+
+
+if __name__ == "__main__":
```

Comment:
These kind of temporary test snippets typically get in the way in the long term. If useful I would move them to standalone example scripts. Otherwise I'd remove it